src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Core.Tests/PluginMetadataSerializationTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿using AutoFixture;
+using Microsoft.Performance.Toolkit.Plugins.Core.Metadata;
+using Microsoft.Performance.Toolkit.Plugins.Core.Serialization;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Fixture = AutoFixture.Fixture;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Core.Tests;
+
+[TestClass]
+public sealed class PluginMetadataSerializationTests
+{
+    private Fixture? fixture = null;
+
+    [TestInitialize]
+    public void Setup()
+    {
+        this.fixture = new Fixture();
+    }
+
+    [TestMethod]
+    public void PluginMetadata_Serialization_RoundTrip()
+    {
+        ISerializer<PluginMetadata> metadataSerializer = SerializationUtils.GetJsonSerializerWithDefaultOptions<PluginMetadata>();
+
+        PluginMetadata metadata = this.fixture!.Create<PluginMetadata>();
+
+        using var stream = new MemoryStream();
+        metadataSerializer.Serialize(stream, metadata);
+
+        stream.Position = 0;
+
+        PluginMetadata deserializedMetadata = metadataSerializer.Deserialize(stream);
+
+        Assert.IsNotNull(deserializedMetadata);
+        Assert.IsTrue(metadata.Equals(deserializedMetadata));
+    }
+
+    [TestMethod]
+    public void PluginContentsMetadata_Serialization_RoundTrip()
```

Comment:
These two methods are very similar. You can create a helper method that uses a generic parameter to actually run the test.

```cs
private void RunSerializationTest<T>()
{
        ISerializer<T> serializer = SerializationUtils.GetJsonSerializerWithDefaultOptions<PluginMetadata>();

        T original = this.fixture!.Create<T>();

        using var stream = new MemoryStream();
        serializer.Serialize(stream, original);

        stream.Position = 0;

        T deserialized = serializer.Deserialize(stream);

        Assert.IsNotNull(deserialized);
        Assert.IsTrue(original.Equals(deserialized));
}
```

Reply:
Added a helper method